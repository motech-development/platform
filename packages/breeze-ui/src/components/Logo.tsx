import { FC, SVGProps } from 'react';

export interface ILogoProps extends SVGProps<SVGSVGElement> {
  alt: string;
}

const Logo: FC<ILogoProps> = ({ alt, ...rest }) => (
  <svg
    {...rest}
    xmlns="http://www.w3.org/2000/svg"
    viewBox="0 0 345.21 398.61"
    aria-labelledby="logo-alt-text"
  >
    <title id="logo-alt-text">{alt}</title>
    <polygon
      className="fill-current"
      points="172.6 199.31 57.53 132.87 57.53 332.18 0 298.96 0 99.65 172.6 0 345.21 99.65 345.21 298.96 287.67 332.18 287.67 132.87 172.6 199.31"
    />
    <polygon
      className="fill-current"
      points="115.07 365.39 115.07 232.53 172.6 265.74 230.14 232.53 230.14 365.39 172.6 398.61 115.07 365.39"
    />
  </svg>
);

export default Logo;
