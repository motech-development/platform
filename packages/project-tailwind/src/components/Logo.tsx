import { ComponentPropsWithRef } from 'react';

/** Logo component properties */
export interface ILogoProps extends ComponentPropsWithRef<'svg'> {
  /** Alternative information text */
  alt: string;
}

/**
 * Motech Development logo
 *
 * @param props - Component props
 *
 * @returns Logo component
 */
export function Logo({ alt, ...rest }: ILogoProps) {
  return (
    <svg
      {...rest}
      xmlns="http://www.w3.org/2000/svg"
      viewBox="0 0 345.21 398.61"
      aria-labelledby="logo-alt-text"
    >
      <title id="logo-alt-text">{alt}</title>
      <polygon
        className="fill-current"
        points="172.6 199.31 57.53 132.87 57.53 332.18 0 298.96 0 99.65 172.6 0 345.21 99.65 345.21 298.96 287.67 332.18 287.67 132.87 172.6 199.31"
      />
      <polygon
        className="fill-current"
        points="115.07 365.39 115.07 232.53 172.6 265.74 230.14 232.53 230.14 365.39 172.6 398.61 115.07 365.39"
      />
    </svg>
  );
}
