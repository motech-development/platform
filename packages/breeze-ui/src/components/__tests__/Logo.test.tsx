import { render } from '@testing-library/react';
import Logo from '../Logo';

describe('Logo', () => {
  it('should correctly render the logo', () => {
    const { container } = render(<Logo alt="Logo" />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should add additional props', () => {
    const { container } = render(<Logo alt="Logo" className="test-class" />);

    expect(container.firstChild).toHaveClass('test-class');
  });
});
