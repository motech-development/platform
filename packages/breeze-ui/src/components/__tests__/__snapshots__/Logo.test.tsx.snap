// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo should correctly render the logo 1`] = `
<svg
  aria-labelledby="logo-alt-text"
  viewBox="0 0 345.21 398.61"
  xmlns="http://www.w3.org/2000/svg"
>
  <title
    id="logo-alt-text"
  >
    Logo
  </title>
  <polygon
    class="fill-current"
    points="172.6 199.31 57.53 132.87 57.53 332.18 0 298.96 0 99.65 172.6 0 345.21 99.65 345.21 298.96 287.67 332.18 287.67 132.87 172.6 199.31"
  />
  <polygon
    class="fill-current"
    points="115.07 365.39 115.07 232.53 172.6 265.74 230.14 232.53 230.14 365.39 172.6 398.61 115.07 365.39"
  />
</svg>
`;
